packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx

```diff
@@ -82,6 +90,71 @@ const WorkspaceHeader: React.FC = () => {
 
   const [versionAlert, setVersionAlert] = useState(false);
 
+  const [upgradeButtonData, setUpgradeButtonData] = useState<{
+    trialDaysLeft?: number;
+    trialLeftProgress?: number;
+    showUpgradeTrialButton: boolean;
+    showUpgradeDefaultButton: boolean;
+  }>({ showUpgradeTrialButton: false, showUpgradeDefaultButton: true });
+
+  const getCurrentSubscription = useCallback(async () => {
```

Comment:
all of this code should be part of stigg hook and pass as a context function/value

Reply:
What is the benefit in doing so?
Is a pattern that we use or is for performance gain? I'm asking as the current implementation makes at most 1 request to stigg apis even on multiple re-render of the component.

Reply:
Moved to an hook